### PR TITLE
[ch9598] Increase Code Coverage of Webservice Routs

### PR DESCRIPTION
**Pull request by BenGalewsky**
# Problem
There is still limited code coverage of the funcX web service endpoints.

Solution to [Clubhouse Issue 9598](https://app.clubhouse.io/funcx/story/9598/increase-code-coverage-to-all-web-service-endpoints)

# Approach
The funcX routes python file is very big - we had initially followed the same structure for the unit tests which made for a very very big python file. The first step in this PR is to break the tests out by the objects they interact with.

The goal of the story is to have at least minimal, happy path tests for every endpoint in the funcX Webservice. No attempt was made to exercise exception paths or some of the optional paths in the functions.

# How to Test
Take a look at the [CodeCov report](https://codecov.io/gh/funcx-faas/funcx-web-service/src/2de576a56bf2b19329d836249cd3a92c07fb7f9a/funcx_web_service/routes/funcx.py) to verify that each route has been tested.

Note that we uncovered an endpoint called `/addr` - not sure if this is really needed

### CODE REVIEW

**Comment by joshbryan-globus** [comment]
[ch9598]

**Review by sirosen — APPROVED** [review]
I have a few comments, but nothing blocking.

There's one line that I believe should be deleted (looks like a copy-paste mistake), but it would also be caught if we switched to using some real schema validation instead of ad-hoc payload inspection. So it's not critical.

File: tests/routes/test_task_groups.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        mocker.patch(
+            "funcx_web_service.models.tasks.RedisField.__get__",
+            return_value=22)
+
+        mock_redis.ttl = mocker.Mock(return_value=10)
+        task_group = TaskGroup(mock_redis, "123", 22)
+        mock_from_id = mocker.patch.object(TaskGroup, "from_id", return_value=task_group)
+        mock_exists = mocker.patch.object(TaskGroup, "exists", return_value=True)
+
+        result = self.client.get("/api/v1/task_groups/123", headers={"Authorization": "my_token"})
```
[the remark this answers is not in the document]
Comment (BenGalewsky) [reply tests/routes/test_task_groups.py:16]: Yeah - this is better. I replicated this from a more complicated setup in a different project

**Comment by BenGalewsky** [comment]
Thank you so much for the comments and ideas, @sirosen

**Pull request merged**